compiler/rustc_passes/src/weak_lang_items.rs

```diff
@@ -70,7 +70,19 @@ fn verify<'tcx>(tcx: TyCtxt<'tcx>, items: &lang_items::LanguageItems) {
                     tcx.sess.note_without_error("Use `#![feature(default_alloc_error_handler)]` for a default error handler");
                 }
             } else {
-                tcx.sess.err(&format!("language item required, but not found: `{}`", name));
+                tcx
+                    .sess
+                    .diagnostic()
+                    .struct_err(&format!("language item required, but not found: `{}`", name))
+                    .note(&format!("This occurs when a binary crate with `#![no_std]` is compiled for a target that usually gets `{}` from the standard library.", name))
+                    .help(&format!(
+r#"If you want to compile for a specific target_triple that doesnt need `{}`, then make sure to run with `--target target_triple` or add the following to the crate's `.cargo/config` file:
```

Comment:
```suggestion
"you may be able to compile for a target that doesn't need `{}`, specify a target with `--target` or in `.cargo/config`"
```
nit: wording

I don't know if there are other diagnostics where we reference what to do in Cargo, might be worth checking if there is precedent.

Reply:
I searched for other references to `.cargo/config` and didnt find any.
But I did stumble across this earlier: https://github.com/rust-lang/rust/blob/2ca933301183b1e913802bc5710c0cb5218ada28/compiler/rustc_metadata/src/locator.rs#L1151-L1156
Which references what to do in rustup.
It also makes the case that its useful even when the user doesnt have rustup because it gives a hint what you would have to do in the equivalent tool.
In this case it hints that you should probably set the target in your build system of choice.